web/modules/custom/dbcdk_community/src/Form/QuarantineCreateForm.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\dbcdk_community\Form\QuarantineCreateForm.
+ */
+
+namespace Drupal\dbcdk_community\Form;
+
+use DBCDK\CommunityServices\Api\QuarantineApi;
+use DBCDK\CommunityServices\ApiException;
+use DBCDK\CommunityServices\Model\Profile;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+/**
+ * Create a Community Quarantine.
+ */
+class QuarantineCreateForm extends FormBase implements ContainerInjectionInterface {
+
+  /**
+   * The date format the Community Client expects.
+   */
+  const DATE_FORMAT = 'Y-m-d';
+
+  /**
+   * The Community Service Profile object from the username context.
+   *
+   * @var \DBCDK\CommunityServices\Model\Profile $profile
+   */
+  protected $profile;
+
+  /**
+   * The DBCDK Community Service Quarantine API.
+   *
+   * @var \DBCDK\CommunityServices\Api\QuarantineApi $quarantineApi
+   */
+  protected $quarantineApi;
+
+  /**
+   * QuarantineCreateForm constructor.
+   *
+   * @param \DBCDK\CommunityServices\Model\Profile $profile
+   *   The Community Profile we wish to grant a quarantine.
+   * @param \DBCDK\CommunityServices\Api\QuarantineApi $quarantine_api
+   *   The Community Service Quarantine API.
+   */
+  public function __construct(Profile $profile, QuarantineApi $quarantine_api) {
+    $this->profile = $profile;
+    $this->quarantineApi = $quarantine_api;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    try {
+      $profile_filter = [
+        'where' => [
+          'username' => $container->get('request_stack')->getCurrentRequest()->get('username'),
+        ],
+      ];
+      /* @var \DBCDK\CommunityServices\Api\ProfileApi */
+      $profile_api = $container->get('dbcdk_community.api.profile');
+      $profile = $profile_api->profileFindOne(json_encode($profile_filter));
+    }
+    catch (ApiException $e) {
+      \Drupal::logger('DBCDK Community Service')->error($e);
+      // Create an empty Profile instance. The constructor expects a Profile
+      // object and will cause a fatal error if it receives something else and
+      // we would like to display an empty page with a Drupal message in case of
+      // an error.
+      $profile = new Profile();
```

Comment:
Instead of creating an empty Profile we could set `$profile = NULL` and set the default value of `$profile` in the constructor to be NULL as well.

This is supported by PHP and we avoid odd checks for validity based on e.g. username.